### PR TITLE
improve raft module and test

### PR DESCRIPTION
**Pull request by javeme**

### CODE REVIEW

**Comment by javeme** [comment]
error when truncate
```java
2022-01-04 17:49:15 [store-backend-executor33] [ERROR] c.b.h.b.s.r.StoreStateMachine - Failed to execute backend command: SNAPSHOT
java.lang.IllegalStateException: Can't apply command for all store at one time
        at com.google.common.base.Preconditions.checkState(Preconditions.java:531) ~[guava-25.1-jre.jar:?]
        at com.baidu.hugegraph.util.E.checkState(E.java:68) ~[hugegraph-common-2.0.1.jar:2.0.1.0]
        at com.baidu.hugegraph.backend.store.raft.StoreStateMachine.applyCommand(StoreStateMachine.java:166) ~[hugegraph-core-0.12.0.jar:0.12.0.0]
        at com.baidu.hugegraph.backend.store.raft.StoreStateMachine.lambda$onApply$1(StoreStateMachine.java:136) ~[hugegraph-core-0.12.0.jar:0.12.0.0]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_111]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_111]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_111]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_111]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]

```

**Comment by javeme** [comment]
raft test error:
```java
Error:  Tests run: 52, Failures: 3, Errors: 32, Skipped: 0, Time elapsed: 84.22 s <<< FAILURE! - in com.baidu.hugegraph.api.ApiTestSuite
Error:  testAppend(com.baidu.hugegraph.api.IndexLabelApiTest)  Time elapsed: 0.137 s  <<< FAILURE!
java.lang.AssertionError: Response with status 400 and content {"exception":"class com.baidu.hugegraph.exception.ExistedException","message":"The index label 'personByAge' has existed","cause":""} expected:<202> but was:<400>
	at com.baidu.hugegraph.api.IndexLabelApiTest.testAppend(IndexLabelApiTest.java:66)

Error:  testAppend(com.baidu.hugegraph.api.IndexLabelApiTest)  Time elapsed: 0.138 s  <<< ERROR!
com.baidu.hugegraph.HugeException: Failed to list indexlabels

Error:  testCreate(com.baidu.hugegraph.api.IndexLabelApiTest)  Time elapsed: 0.034 s  <<< FAILURE!
java.lang.AssertionError: Response with status 400 and content {"exception":"class com.baidu.hugegraph.exception.ExistedException","message":"The index label 'personByAge' has existed","cause":""} expected:<202> but was:<400>
	at com.baidu.hugegraph.api.IndexLabelApiTest.testCreate(IndexLabelApiTest.java:52)

Error:  testCreate(com.baidu.hugegraph.api.IndexLabelApiTest)  Time elapsed: 0.034 s  <<< ERROR!
com.baidu.hugegraph.HugeException: Failed to list indexlabels

```

**Comment by javeme** [comment]
how to upload multiple report files to codecov:
https://github.com/codecov/codecov-action/issues/47

**Review by Linary — APPROVED** [review]

File: hugegraph-test/src/main/java/com/baidu/hugegraph/api/GremlinApiTest.java [hunk 1]
```diff
@@ -86,20 +87,37 @@ public void testScript() {
...
+        body = "{"
                 + "\"gremlin\":\""
-                + "def auth = hugegraph.hugegraph().authManager();"
-                + "def admin = auth.findUser('admin');"
-                + "hugegraph.clearBackend();"
-                + "hugegraph.initBackend();"
-                + "auth.createUser(admin);\","
+                + "  if (!hugegraph.backendStoreFeatures()"
+                + "                .supportsSharedStorage())"
+                + "    return;"
```
Comment (imbajin) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/api/GremlinApiTest.java:107]: merge to one line maybe?

File: hugegraph-test/src/main/java/com/baidu/hugegraph/api/GremlinApiTest.java [hunk 2]
```diff
@@ -86,20 +87,37 @@ public void testScript() {
...
+                + "  auth.createUser(admin);\","
                 + "\"bindings\":{},"
                 + "\"language\":\"gremlin-groovy\","
                 + "\"aliases\":{\"g\":\"__g_hugegraph\"}}";
         assertResponseStatus(200, client().post(path, body));
 
         body = "{"
                 + "\"gremlin\":\"hugegraph.serverStarted("
-                + "IdGenerator.of('server1'), NodeRole.MASTER)\","
+                + "              IdGenerator.of('server1'), NodeRole.MASTER)\","
```
Comment (imbajin) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/api/GremlinApiTest.java:120]: same

**Review by imbajin — APPROVED** [review]

**Pull request merged**